### PR TITLE
dbeaver/pro# 2502 Impossible to delete project pass without entering password  

### PR DESCRIPTION
**Pull request by serjiokov**
Fix behavior clear password button

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceSerializerModern.java [hunk 1]
```diff
@@ -829,7 +829,7 @@ private Map<String, Map<String, Map<String, String>>> readSecureCredentials(
                 }
             }
             log.error("Error reading secure credentials", e);
-            throw new DBException("Project configuration can not be open", e);
+            return null;
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceSerializerModern.java]: I'm not sure about this change. Haven't we have the case when changing error to null broke something?

Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceSerializerModern.java]: Sure it looks a bit strange but we have to return null in order to proceed open datasources (requirement) with lost credentials

Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceSerializerModern.java]: Im rethinking a bit of approach, added  specific condition.

**Comment by serjiokov** [comment]
Worked as disscussed (user able to see connections with lost credentials):
image

**Comment by serjiokov** [comment]
In CE application
image

File: plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceSerializerModern.java [hunk 2]
```diff
@@ -384,15 +386,30 @@ public boolean parseDataSources(
...
-
+        if (CommonUtils.toBoolean(registry.getProject().getProjectProperty(USE_PROJECT_PASSWORD))) {
+            if (!DBWorkbench.getPlatform().getApplication().isHeadlessMode()
+                && DBWorkbench.getPlatform().getApplication().isCommunity()) {
+                if (DBWorkbench.getPlatformUI().confirmAction(
+                    RegistryMessages.project_open_cannot_read_credentials_title,
+                    NLS.bind(RegistryMessages.project_open_cannot_read_credentials_message,
+                        registry.getProject().getName()),
+                    RegistryMessages.project_open_cannot_read_credentials_button_text, true)) {
+                    //
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceSerializerModern.java]: empty comment?

Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/DataSourceSerializerModern.java]: fixed

**Comment by MashaKorax** [comment]
1. Install project pass + encrypt configuration checkbox
2. Add at least 1 connection to the project
3. Re-login to beaver
4. Go to project settings, click clear password
5. **Enter the correct password**

**The actual result** is that the project will open, and connections will disappear
**As expected** - the project will open, connections are in place along with credentials

**Comment by MashaKorax** [comment]
verified

**Pull request merged**